grafana-plugin/src/containers/ServiceNowConfigDrawer/CompleteServiceNowConfigModal.tsx

```diff
@@ -0,0 +1,134 @@
+  onHide: () => void;
+}
+
+interface FormFields {
+  additional_settings: ApiSchemas['AlertReceiveChannel']['additional_settings'];
+}
+
+export const CompleteServiceNowModal: React.FC<CompleteServiceNowConfigModalProps> = ({ onHide }) => {
+  const { alertReceiveChannelStore } = useStore();
+  const integration = useCurrentIntegration();
+
+  const formMethods = useForm<FormFields>({
+    values: {
+      additional_settings: {
+        ...integration.additional_settings,
+      },
+    },
+  });
+
+  const [isFormActionsDisabled, setIsFormActionsDisabled] = useState(false);
+
+  const styles = useStyles2(getStyles);
+  const { handleSubmit } = formMethods;
+
+  const { id } = integration;
+
+  return (
+    <Modal closeOnEscape={false} isOpen title={'Complete ServiceNow configuration'} onDismiss={onDismiss}>
+      <FormProvider {...formMethods}>
+        <form onSubmit={handleSubmit(onFormSubmit)}>
+          <div className={styles.border}>
+            <ServiceNowStatusSection />
+          </div>
+
+          <div className={styles.border}>
+            <ServiceNowTokenSection />
+          </div>
+
+          <div>
+            <HorizontalGroup justify="flex-end">
+              <Button variant="secondary" onClick={onFormAcknowledge} disabled={isFormActionsDisabled}>
+                Close
+              </Button>
+              <Button variant="primary" type="submit" disabled={isFormActionsDisabled}>
+                Proceed
+              </Button>
+            </HorizontalGroup>
+          </div>
+        </form>
+      </FormProvider>
+    </Modal>
+  );
+
+  async function onDismiss() {
+    setIsFormActionsDisabled(true);
+    try {
+      await onFormAcknowledge();
+      onHide();
+    } catch (ex) {
+      setIsFormActionsDisabled(false);
+    }
+  }
+
+  async function onFormAcknowledge() {
+    setIsFormActionsDisabled(true);
+
+    try {
+      await alertReceiveChannelStore.update({
+        id,
+        data: {
+          ...integration,
+          additional_settings: {
+            // use existing fields
+            ...integration.additional_settings,
+            is_configured: true,
+          },
+        },
+      });
+
+      onHide();
+    } catch (ex) {
+      setIsFormActionsDisabled(false);
+    }
+  }
+
+  async function onFormSubmit(formData: FormFields) {
+    setIsFormActionsDisabled(true);
+
+    const data: OmitReadonlyMembers<ApiSchemas['AlertReceiveChannel']> = {
+      ...integration,
+      ...formData,
+
+      additional_settings: {
+        ...integration.additional_settings,
+        ...formData.additional_settings,
+        state_mapping: {
+          ...formData.additional_settings.state_mapping,
+        },
+        is_configured: true,
+      },
+    };
+
+    try {
+      await alertReceiveChannelStore.update({ id, data });
+      openNotification('You successfully completed your ServiceNow configuration');
+      onHide();
+    } finally {
+      setIsFormActionsDisabled(false);
+    }
+  }
+};
```

Comment:
Instead of doing `try...catch...` and managing `isFormActionsDisabled` local state in this component, we can make use of existing `ActionKey.UPDATE_INTEGRATION` loading state and based on that disable the buttons.
```
const isLoading = useIsLoading(ActionKey.UPDATE_INTEGRATION);
...

              <Button variant="secondary" onClick={onFormAcknowledge} disabled={isLoading}>
                Close
              </Button>
              <Button variant="primary" type="submit" disabled={isLoading}>
                Proceed
              </Button>
            </HorizontalGroup>
```

Then remove`useState` usage

Reply:
The integration update seems too generic for me and I don't want to couple it to the loading state of this modal.